### PR TITLE
ENH: REACH data

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

- Adds reach dosimeter data
- Fixes docstring copy paste issues

# Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

```
import datetime as dt
import pysat
import pysatNASA

reach = pysat.Instrument(inst_module=pysatNASA.instruments.reach_dosimeter, inst_id='101')
reach.remote_file_list()

reach.download(dt.datetime(2017, 4, 10))
reach.load(date=dt.datetime(2017, 4, 10))
reach.data

```

**Test Configuration**:
* Operating system: Ventura
* Version number: Python 3.9.12

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
Still needs:
- [ ] acknowledgement string (can I import this from cdasws?)
- [ ] reference string
- [ ] double-check meta translation

**Comment by jklenzing** [comment]
The acknowledgement string is more complicated than expected. Using standard copy paste for now, documented potential paths forward in #207.

**Comment by jklenzing** [comment]
Note for testing:
```
import pysat
import pysatNASA
reach = pysat.Instrument(inst_module=pysatNASA.instruments.reach_dosimeter, inst_id='101')
reach.acknowledgements
```
yields
```
'Please load a file for full acknowledgments.'
```
Assuming you have downloaded files, running
```
reach.load(date=reach.files.files.index[0])
reach.acknowledgements
```
will then yield the full acknowledgement string.

File: pysatNASA/instruments/methods/reach.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+                             "AEROSPACE REPORT NO. TOR-2019-02361")),
+                   ' '.join(("Halford, A.J., Guild, T., O'Brien, T.P., Boyd,",
+                             "A.J., Mazur, J.E. (2019)",
+                             "REACH Maps and Indices for UDL: Version 1,",
+                             "AEROSPACE REPORT NO. TOR-2019-02650")),
+                   ' '.join(("Guild, T., O'Brien, T.P., Boyd, A.J., Mazur,",
+                             "J.E. (2021)",
+                             "REACH Maps and Indices for UDL: Version 1",
+                             "AEROSPACE REPORT NO. TOR-2021-01076"))
+                   ))}
```
Comment (aburrell) [inline pysatNASA/instruments/methods/reach.py:19]:
```suggestion
                             "AEROSPACE REPORT NO. TOR-2021-01076"))))}
```

File: pysatNASA/instruments/reach_dosimeter.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    Note
+    ----
+    Any additional keyword arguments passed to pysat.Instrument
+    upon instantiation are passed along to this routine.
+
+    Examples
+    --------
+    ::
+
+        inst = pysat.Instrument('icon', 'ivm', inst_id='a', tag='')
```
Comment (aburrell) [inline pysatNASA/instruments/reach_dosimeter.py:127]: Fix example

File: pysatNASA/instruments/reach_dosimeter.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    Parameters
+    ----------
+    fnames : array-like
+        iterable of filename strings, full path, to data files to be loaded.
+        This input is nominally provided by pysat itself.
+    tag : string
+        tag name used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself.
+    inst_id : string
```
Comment (aburrell) [inline pysatNASA/instruments/reach_dosimeter.py:107]:
```suggestion
    tag : str
        tag name used to identify particular data set to be loaded.
        This input is nominally provided by pysat itself.
    inst_id : str
```

File: pysatNASA/instruments/reach_dosimeter.py (hunk 2 above)
Comment (jklenzing) [inline pysatNASA/instruments/reach_dosimeter.py:127]:
```suggestion
        inst = pysat.Instrument('reach', 'dosimeter', inst_id='101', tag='')
```

**Comment by aburrell** [comment]
Ok, the data loads well, but the global file attributes in MetaHeader have problems.  A benign example that I would let pass is  `reach.meta.header.Notes` is `nan`.  A malignant example is `Temperature_Sensitivity_43-65_deg`, whose string format is not appropriate for an attribute name and so is not accessible.  I think this can be fixed by replacing the `-` with a `_` in the load function.

**Review by aburrell — CHANGES_REQUESTED** [review]
Fix load routine so that all meta header attributes are accessible using standard attribute access method.

File: pysatNASA/instruments/reach_dosimeter.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+inst_id
+    '101', '102', '105', '108', '113', '114', '115', '116', '133', '134', '135',
+    '136', '137', '138', '139', '140', '148', '149', '162', '163', '164', '165',
+    '166', '169', '170', '171', '172', '173', '175', '176', '180', '181'
+
+
+"""
+
+import datetime as dt
+import functools
```
Comment (jklenzing) [inline pysatNASA/instruments/reach_dosimeter.py:33]:
```suggestion
import functools
import numpy as np
```

**Review by aburrell — APPROVED** [review]

**Pull request merged**